1_fetch/src/get_sb_data.R

```diff
@@ -272,5 +205,28 @@ calc_avg_wildfire <- function(sb_data) {
     arrange(COMID, label) %>%
     select(COMID, label, avg_annual) %>%
     pivot_wider(names_from = "label", values_from = "avg_annual")
-  return(wildfire)
-}
\ No newline at end of file
+  
+  data <- data %>%
+    select(-ends_with(".y"), -ID, -starts_with("..."), -contains("_S1"), -contains("PHYSIO_AREA"), 
+           -contains("SOHL"), -contains("NDAMS"), -contains("STORAGE"), 
+           -contains("MAJOR"), -contains("TAV"), -contains("PPT"), -contains("WILDFIRE")) %>%
```

Comment:
There are a few TAV and PPT columns we'll want to retain (min, max, average annual). I think filtering with `_TAV_` and `_PPT_` instead works to only remove the monthly TAV and PPT columns

Reply:
Good catch!

Reply:
There are some other features that could be reduced here. Also happy to do these in a separate step. There are some Nitrogen inputs, NLCD land cover (or maybe you already removed that SB link?) 

# Nitrogen
'NO10AVE', 'NO200AVE', 'NO4AVE', 
# Land cover attributes that are 1 yr, few years, or NLCD
'NLCD', 'IMPV',  'ESTUARY',  'CNPY', 'WRSE'
# physiographic area coverage is not useful - percent of catchment covered by physiographic regions.
'PHYSIO_AREA',
# this runoff is by a coarse HUC level, not reach level. We have a different water balance model runoff that's by reach, so we can drop this one.
'RUN7100'
# Catchment soils - these have NA that are usually water, but somehow the TOT/ACC attributes do not have NAs, which makes me suspicious of the CAT NAs. I'm opting to remove them and use TOT or ACC. 
"CAT_HGA", "CAT_HGAC", "CAT_HGAD", "CAT_HGB", "CAT_HGBC", "CAT_HGBD", "CAT_HGC", "CAT_HGCD", "CAT_HGD", "CAT_HGVAR"

Reply:
> Nitrogen
'NO10AVE', 'NO200AVE', 'NO4AVE',

These are in the [STATSGO Soil Characteristics, Layer Attributes](https://www.sciencebase.gov/catalog/item/5728decfe4b0b13d3918a9aa?community=National+Water-Quality+Assessment+-+Hydrologic+Systems+Team+%28NAWQA+-+HST%29). Probably will be highly correlated to other soil attributes, but that is probably best to be determined in your feature engineering steps. 

> Land cover attributes that are 1 yr, few years, or NLCD
'NLCD', 'IMPV', 'ESTUARY', 'CNPY', 'WRSE'

Already removed from sb_var_ids.csv - we can add these back in as needed if we decide to use NLCD later. 

> physiographic area coverage is not useful - percent of catchment covered by physiographic regions.
'PHYSIO_AREA',

Already included in line 210. 

> this runoff is by a coarse HUC level, not reach level. We have a different water balance model runoff that's by reach, so we can drop this one.
'RUN7100'

Already removed from sb_var_ids.csv.

> Catchment soils - these have NA that are usually water, but somehow the TOT/ACC attributes do not have NAs, which makes me suspicious of the CAT NAs. I'm opting to remove them and use TOT or ACC.
"CAT_HGA", "CAT_HGAC", "CAT_HGAD", "CAT_HGB", "CAT_HGBC", "CAT_HGBD", "CAT_HGC", "CAT_HGCD", "CAT_HGD", "CAT_HGVAR"

Do you think it would be better to remove these in your later feature engineering steps? I think all of the decisions regarding selections between CAT / ACC / TOT are better suited for the model prep and feature selection stage (even if we know up-front that they might not be useful).

Reply:
Yes, the remaining attributes can be removed in a separate target for feature selection